### PR TITLE
QuickStart template for Azure Batch pool without public IP addresses

### PR DESCRIPTION
**Pull request by lilinvictorms**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add quickstart template for Azure Batch pool without public IP addresses

### CODE REVIEW

**Comment by lilinvictorms** [comment]
@bmoore-msft could you please review?

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        '10.0.0.0/16'
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: '10.0.0.0/24'
+          networkSecurityGroup: {
+            id: resourceId('Microsoft.Network/networkSecurityGroups', nsgName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:89]: use bicep syntax

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        }
+      }
+    ]
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2022-01-01' = {
+  name: vnetName
+  location: location
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:74]: remove dependsOn

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  name: '${privateDnsZoneName}/${vnetName}'
+  location: 'global'
+  dependsOn: [
+    privateDnsZone
+    vnet
+  ]
+  properties: {
+    registrationEnabled: false
+    virtualNetwork: {
+      id: resourceId('Microsoft.Network/virtualNetworks', vnetName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:108]: use bicep syntax

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        }
+      }
+    ]
+  }
+}
+
+resource vnetDnsZoneLink 'Microsoft.Network/privateDnsZones/virtualNetworkLinks@2020-06-01' = {
+  name: '${privateDnsZoneName}/${vnetName}'
+  location: 'global'
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:101]: remove dependsOn

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    batchAccount
+    vnet
+  ]
+  properties: {
+    privateLinkServiceConnections: [
+      {
+        name: nodeManagementPrivateEndpointName
+        properties: {
+          privateLinkServiceId: resourceId('Microsoft.Batch/batchAccounts', accountName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:125]: use bicep syntax

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+    virtualNetwork: {
+      id: resourceId('Microsoft.Network/virtualNetworks', vnetName)
+    }
+  }
+}
+
+resource nodeManagementPrivateEndpoint 'Microsoft.Network/privateEndpoints@2022-01-01' = {
+  name: nodeManagementPrivateEndpointName
+  location: location
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:116]: remove dependsOn

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        properties: {
+          privateLinkServiceId: resourceId('Microsoft.Batch/batchAccounts', accountName)
+          groupIds: [
+            'nodeManagement'
+          ]
+        }
+      }
+    ]
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', vnetName, subnetName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:133]: use bicep syntax

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+  dependsOn: [
+    nodeManagementPrivateEndpoint
+    privateDnsZone
+  ]
+  properties: {
+    privateDnsZoneConfigs: [
+      {
+        name: 'private-dns-zone-integration'
+        properties: {
+          privateDnsZoneId: resourceId('Microsoft.Network/privateDnsZones', privateDnsZoneName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:149]: use bicep syntax

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+    ]
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', vnetName, subnetName)
+    }
+  }
+}
+
+resource privateEndpointDnsIntegration 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2020-11-01' = {
+  name: '${nodeManagementPrivateEndpointName}/default'
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:140]: remove dependsOn

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+          privateDnsZoneId: resourceId('Microsoft.Network/privateDnsZones', privateDnsZoneName)
+        }
+      }
+    ]
+  }
+}
+
+resource batchPool 'Microsoft.Batch/batchAccounts/pools@2022-06-01' = {
+  name: '${accountName}/${poolName}'
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:158]: remove dependsOn

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+          publisher: 'canonical'
+          offer: 'ubuntuserver'
+          sku: '18.04-lts'
+          version: 'latest'
+        }
+        nodeAgentSkuId: 'batch.node.ubuntu 18.04'
+      }
+    }
+    networkConfiguration: {
+      subnetId: resourceId('Microsoft.Network/virtualNetworks/subnets', vnetName, subnetName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:182]: use bicep syntax

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep [hunk 12]
```diff
@@ -0,0 +1,195 @@
+@description('Name of the Batch account')
+param accountName string = 'mytestaccount'
```
[the remark this answers is not in the document]
Comment (lilinvictorms) [reply quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep]: We do need unique resource name. Removed default value and keep it in the parameters.

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/azuredeploy.parameters.json [hunk 13]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "accountName": {
```
[the remark this answers is not in the document]
Comment (lilinvictorms) [reply quickstarts/microsoft.batch/batch-pool-no-public-ip/azuredeploy.parameters.json:5]: We do need customer to provide these 2 resource names. Removed their default values from bicep, and keep them here in the parameters file.

File: quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep (hunk 10 above)
Comment (lilinvictorms) [reply quickstarts/microsoft.batch/batch-pool-no-public-ip/main.bicep:145]: Keep dependency on private endpoint as the pool requires it to work normally.

**Pull request merged**